src/services/catalog.service.ts

```diff
@@ -96,25 +96,39 @@ export class CatalogService {
     return new TemplateItem(resource, this);
   }
 
-  public getCategoriesBySubCategories(tags: any) {
+  public getCategoriesBySubCategories(itemTags: any) {
     let catsBySubCats = {};
     let otherId = 'other';
-
-    _.each(tags, (tag) => {
-      _.each(this.categories, (category) => {
-        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
-              return subCategory.id === tag || _.includes(subCategory.categoryAliases, tag);
-            }
-        );
-        if (subCat) {
-          catsBySubCats[subCat.id] = category.id;
-          return false;
+    let filteredSubCats: any;
+
+    _.each(this.categories, (category) => {
+      if (category.tags) {
+        if (!_.isEmpty(this.getMatchingTags(category.tags, itemTags))) {
+          filteredSubCats = this.filterSubCatsByTags(category.subCategories, itemTags);
+          if (!_.isEmpty(filteredSubCats)) {
+            _.each(filteredSubCats, (subCat) => {
+              catsBySubCats[subCat.id] = category.id;
+            });
+          } else {
+            catsBySubCats[otherId] = category.id;
```

Comment:
this is going to be broken if you end up categorizing an item into more than one main category's "Other", i end up in Foo -> Other and Bar -> Other, the second one that matches wins, where we really want it to be in both

I think in general we should not be assuming that subcategory id's are unique across main categories, a subcategory id just needs to be unique within its main category